migrations/versions/48ee74b8a7c8_.py

```diff
@@ -0,0 +1,141 @@
+            token_realms = TokenRealm.query.filter(TokenRealm.token_id == token.id).all()
+            realm_id = None
+            if not token_realms:
+                print(u"Error while migrating token assignment for token {0!s}. Missing Realm!".format(token.serial))
+            elif len(token_realms) == 1:
+                realm_id = token_realms[0].realm_id
+            elif len(token_realms) > 1:
+                # If the resolver is only contained in one realm, we fetch the realms:
+                reso_realms = ResolverRealm.query.filter(ResolverRealm.resolver == token.resolver).all()
+                if not reso_realms:
+                    sys.stderr.write(u"The token is {0!s} assigned, but the assigned resolver {1!s} is not "
+                                     u"contained in any realm!".format(token.id, token.resolver))
+                elif len(reso_realms) == 1:
+                    # The resolver is only in one realm, so this is the new realm of the token!
+                    realm_id = reso_realms[0].realm_id
+                elif len(reso_realms) > 1:
+                    # The resolver is contained in two realms, we have to apply more logic between the realms in which
+                    # the resolver is contained and the realms, to which the token is assigend.
+                    found_realm_ids = []
+                    for token_realm in token_realms:
+                        if token_realm.realm_id in [r.realm_id for r in reso_realms]:
+                            # The token realm, that also fits the resolver_realm is used as owner realm
+                            found_realm_ids.append(realm_id)
+                        if len(found_realm_ids) > 1:
+                            sys.stderr.write(u"Your realm configuration for token {0!s} is not distinct!. "
+                                             u"The tokenowner could be in multiple realms! "                                  
+                                             u"The token is assigned to the following realms and the resolver is also "
+                                             u"contained in these realm IDs: {1!s}.".format(token.id, found_realm_ids))
+                        elif len(found_realm_ids) == 1:
+                            realm_id = found_realm_ids[0]
+                        else:
+                            sys.stderr.write(u"Can not assign token {0!s}. The resolver is not contained in any "
+                                             u"realms, to which the token is assigned!".format(token.id))
+
+            to = TokenOwner(token_id=token.id, user_id=token.user_id,
+                            resolver=token.resolver, resolver_type=token.resolver_type,
+                            realm_id=realm_id)
+            session.add(to)
+        session.commit()
+
+    except Exception as exx:
+        print("Failed to migrate token assignment data!")
+        print (exx)
```

Comment:
This should probably also drop the unused columns from ``Token``, if the data has been migrated successfully.

Reply:
OK, I addressed this thing in 72f79b2e

Reply:
What happend to this f* PR?
I can not comment on your comment. (now I can) :-(
Well, a file would be nice, but on the other hand, the admin created a backup, right?
To which file should this be dumped?